sotodlib/preprocess/preprocess_util.py

```diff
@@ -295,8 +295,9 @@ def load_preprocess_det_select(obs_id, configs, context=None,
     pipe = Pipeline(configs["process_pipe"], logger=logger)
 
     meta = context.get_meta(obs_id, dets=dets, meta=meta)
-    logger.info(f"Cutting on the last process: {pipe[-1].name}")
-    pipe[-1].select(meta)
+    logger.info("Restricting detectors on all processes")
+    for process in pipe[:]:
+        process.select(meta)
```

Comment:
I think since you fixed this here, you don't need to run the select step `load_and_preprocess` so on line 347 you should be able to run `pipe.run` with `select=False` and if other functions call `load_preprocess_det_select` before `pipe.run` then the `select=False` should be used there as well. But we should test that it doesn't break other things. 

It would be awesome if `load_and_preprocess_det_select` could collect all of the restriction flags and union them and do one call the restrict as there is quite an overhead to each call to restrict (hence my suggestion to not call it again in pipe.run).

Reply:
Sounds like a good plan.